src/pydocstyle/checker.py

```diff
@@ -498,22 +513,24 @@ def _check_section_underline(cls, section_name, context, indentation):
                                       section_name,
                                       len(dash_line.strip()))
 
-            line_after_dashes = \
-                context.following_lines[next_non_empty_line_offset + 1]
-            if not is_blank(line_after_dashes):
-                yield violations.D410(section_name)
-
             if leading_space(dash_line) > indentation:
                 yield violations.D215(section_name)
 
+            if next_non_empty_line_offset + 1 < len(context.following_lines):
```

Comment:
I changed some of the logic here. The method is a bit long, but I don't think there's a non-artificial way of splitting it to several ones. Tell me if you think I should split it anyways.